src/components/nodes/boundaryTimerEvent/boundaryTimerEvent.vue

```diff
@@ -87,15 +93,16 @@ export default {
 
     if (task) {
       task.embed(this.shape);
-      const { x, y } = snapToAnchor(this.shape.position(), task);
-      const { width } = this.shape.size();
-
-      this.shape.position(x - (width / 2), y - (width / 2));
+      this.updateSnappingPosition(task);
       this.renderBoundaryTimer(this.node.definition.cancelActivity);
-      this.shape.set('elementMove', false);
     } else {
       this.$emit('remove-node', this.node);
     }
+
+    this.shape.listenTo(this.paper, 'element:pointerup', () => {
+      this.updateSnappingPosition(task);
```

Comment:
Since you're passing in task, should this not be called within the `if (task)` block?